### DISCUSSION TITLE
How to extract variables from the name of the PDF file. Then include the variables in an HL7 message.

### DISCUSSION THREAD

**Discussion by guillermoalv91** [nextgenhealthcare/connect#5953] [Q&A] [answered]
I need to create a channel that will receive PDF files from a directory in the source (file reader). The idea is to extract 4 variables from the name of the PDF, example: "Name_Surname_Date_PatientID.pdf". I would later use these variables to complete an HL7 ORU "template" message (with the PDF encapsulated in base64 in the OBX segment) at the destination. 
Can someone tell me, at least the first part to extract the name data from the PDF and save it so I can use it later in the channel?I would like it to include a logger.info to verify that the variables are saved. 
I understand that the "source" must be a File reader and the file type would be binary? How should I configure the Data types (inbound and outbound) section? Where should I place the javascript code? like a transformer at the origin? in the "scripts" tab of the channel? I'm a bit lost currently.

Thanks in advance.

**Comment by pacmano1** [comment, marked as answer]
You can call a .split() and .slice() in javascript to split strings to an array for further referencing.  The file name is referencable in a transformer with:

// these do the same exact thing, get the value from the source map.
`$s('originalFilename')`
`sourceMap.get('originalFilename')`

So you can put your javascript code in a filter or a transfomer on either the source or destination tabls and store those in map variables for subsequent use.

Yes, your file reader would be binary.  Also you might consider using the attachment handler.   See https://forums.mirthproject.io/articles/175046-using-attachments

(a reminder that "thanking in advance" is a contentious expression, perhaps best not to use it).

**Reply by guillermoalv91** [reply]
I have already been able to save the PDF name data in the variables. 

Thank you so much.
